sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -158,6 +167,57 @@ public FormRecognizerServiceVersion getServiceVersion() {
             fetchExtractReceiptResult(includeTextDetails));
     }
 
+
+    /**
+     * List all available models.
+     *
+     * @return {@link PagedFlux} of {@link CustomFormModelInfo}.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<CustomFormModelInfo> listModels() {
+         try {
+            return new PagedFlux<>(() -> withContext(context -> listFirstPageModelInfo(context)),
+                continuationToken -> withContext(context -> listNextPageModelInfo(continuationToken, context)));
+        } catch (RuntimeException ex) {
+            return new PagedFlux<>(() -> monoError(logger, ex));
+        }
+    }
+
+    PagedFlux<CustomFormModelInfo> listModels(Context context) {
+        return new PagedFlux<>(() -> listFirstPageModelInfo(context),
+            continuationToken -> listNextPageModelInfo(continuationToken, context));
+    }
+
+
+    private Mono<PagedResponse<CustomFormModelInfo>> listFirstPageModelInfo(Context context) {
+        return service.listCustomModelsSinglePageAsync(context)
+            .doOnRequest(ignoredValue -> logger.info("Listing all model information"))
+            .doOnSuccess(response -> logger.info("Listed all models"))
+            .doOnError(error -> logger.warning("Failed to list all models", error))
+            .map(res -> new PagedResponseBase<>(
+                res.getRequest(),
+                res.getStatusCode(),
+                res.getHeaders(),
+                toCustomFormModelInfo(res.getValue()),
+                res.getContinuationToken(),
+                null));
+    }
+
+    private Mono<PagedResponse<CustomFormModelInfo>> listNextPageModelInfo(String nextPageLink, Context context) {
+        return service.listCustomModelsNextSinglePageAsync(nextPageLink)
+            .doOnSubscribe(ignoredValue -> logger.info("Retrieving the next listing page - Page {}", nextPageLink))
```

Comment:
Should mention about models info here?

Reply:
I feel it is more about the next page link. So we should not mention about the models info here.